### DISCUSSION TITLE
ExtrudeFromSlices with slices with different number of vertices

### DISCUSSION THREAD

**Discussion by napter** [jscad/OpenJSCAD.org#959] [Q&A]
When using extrude slices with slices that have different number of vertices doesn't work. Here is an example of what it does (the circle on top has more segments than the circle on bottom) 

![image](https://user-images.githubusercontent.com/6260841/144541921-7e368563-a99f-48b0-80c8-911f501534cc.png)

Perhaps this is expected. I do see in the code that it is attempting to do _something_ when the two slices don't match but I am not sure what and if it meant to handle this scenario.

**Comment by hrgdavor** [comment]
Looks like it needs more testing and fixing for sure.

**Reply by z3dev** [reply]
The jump is too large… try small increments of change.

The internal algorithm tries to stitch the sides together but everything depends on the alignment of points. The controlling function has to control the amount of change.

this is mentioned in the user guide as well but no one reads the user guide. 😔

**Comment by hrgdavor** [comment]
Personally, if I would implement extrude from slice, I would put constraints on the slices to simplify the alg. The developer supplying the slices should take care each slice is compatible with the previous.

I can investigate if this is 
 - a bug in extrude
 - "bad" data, but something that extrude alg can resolve automatically
 - "bad" data that needs to be fixed before using for extrude

What is needed thugh is a test script that demonstrates this issue on in the photo

Even if we can significantly improve extrude to be more robust with odd shapes it will take time. We need to know learn to make slices better in the meantime and give some tutorials on possible pitfals.
